contracts/src/rollup/RollupCore.sol

```diff
@@ -411,18 +413,18 @@ abstract contract RollupCore is IRollupCore, PausableUpgradeable {
                 // and cannot go backwards
                 require(afterInboxPosition >= prevInboxPosition, "ERRORED_INBOX_TOO_FEW");
             } else if (assertion.afterState.machineStatus == MachineStatus.FINISHED) {
-                // Assertions must consume exactly all inbox messages
-                // that were in the inbox at the time the previous assertion was created
-                require(
-                    afterInboxPosition == assertion.beforeStateData.configData.nextInboxPosition, "INCORRECT_INBOX_POS"
```

Comment:
need to check `afterInboxPosition <= assertion.beforeStateData.configData.nextInboxPosition` otherwise you can [force it into a valid error state](https://github.com/OffchainLabs/challenge-protocol-v2/blob/1d60251fa1ddf7404d07ce99bd4e112df8a8f058/contracts/src/osp/OneStepProverHostIo.sol#L215-L221) by reading oob

Reply:
hmm, maybe not because [this](https://github.com/OffchainLabs/challenge-protocol-v2/blob/1d60251fa1ddf7404d07ce99bd4e112df8a8f058/contracts/src/osp/OneStepProofEntry.sol#L96-L98) should prevent the machine being kickstarted when all messages are read